### DISCUSSION TITLE
Ability to put gamer servers on separate drives on the same machine.

### DISCUSSION THREAD

**Discussion by Kriz02** [gameserverapp/Platform#173] [Suggestions]
Will  help with prioritizing serer loads when using a higher core machine to run more servers at once.

**Comment by mvaessen** [comment]
I'm currently working on DediConnect suggestions and wanted to invite people to discuss this suggestion.

**This is an invitation to other to join the conversation to discuss how we're going to implement this!**

---

There are various ways how this can be implemented on GSA, where some options take a lot more time to implement vs other solutions. 

The easiest solution is to basically have everything installed on disk A or B. There would be no option to change this later without reinstalling the machine and everything is installed on the configured disk. And docker images would probably still get installed on the main drive, as this is default for Docker.

Since most games are not bottlenecked by the disk is this mostly to allow people separating the boot disk from the game server disk.

---

Options where each game server can be installed on separate drives does require a bit more work, but should make it easier for people to set it up however they prefer. Moving to another disk would require a reinstall (per game server).  As with the first option would docker images probably still be stored on the main drive.

**Comment by LimitlessDrag0n** [comment]
Sorry, if this is a stupid question but what exactly is meant by "the docker image still be stored on main drive" does it mean that the game servers we install, keep the 30GB (for example) or just the installation iso (don't know if worded correctly)?

The second option gives more freedom for admins. But on second thought, the first one seems reasonable as well :D 

My thought: Are those options for running systems or just fresh installs? 
Maybe a "move" option like the following would also be an option? 

![image](https://github.com/gameserverapp/Platform/assets/99832504/9dbb774f-a772-4b55-ba48-840836525153)

**Reply by mvaessen** [reply]
You know there are no such things as stupid questions :)

For windows specifically GSA downloads the game files to `C:\GameServerApp\games` and docker images, which are 10GB each for Windows containers, are handled by docker. So docker receives instructions to download X and it will determine where it's installed. I think this is installed in `Program files` or `ProgramData` on Windows.

So while selecting `D:` as the default install location in this example may still result in Docker images being downloaded to the `C:`. So that's not ideal, but IMO not a problem.

Option 1 is something that will be selected during install of the machine. So it cant be changed unless you reinstall the machine.
Option 2 is determined on a per-game server basis, which would mean you can change the disk per game server while the machine is running. This does require a game server reinstall.

**Reply by LimitlessDrag0n** [reply]
Okay, so IMO Option 2 would be better. In my case, I could store all images on C: and move the gameservers to D: without a reinstall.
Also for those who already have a good running service running through you with multiple machines and servers would be better. Selfhosters, are able to just add a drive without reinstalling the machine in total

**Comment by mvaessen** [comment]
This is now available on the dashboard! Thanks for your suggestions
